### ISSUE TITLE
ajv.validateSchema does not validate properly for schemas using $data

### ISSUE THREAD

**Issue by EtaiG** [ajv-validator/ajv#2275]
**What version of Ajv are you using? Does the issue happen if you use the latest version?**
draft-07

Note that in the example I did not require draft-07 explicitly, but I can set the `meta: false` option and add the draft-07 option explicitly - the same issue occurs

```javascript
const Ajv = require('ajv');
const dataMetaSchema = require('ajv/lib/refs/data.json')

const ajv = new Ajv();
ajv.addMetaSchema(dataMetaSchema)

const schema = {
    type: 'object',
    properties: {
        a: {type: 'number'},
        b: {type: 'number', minimum: {$data: '/a'}}
    }
};


const isValid = ajv.validateSchema(schema);

console.log(isValid);
console.log(ajv.errors);
```

**Validation result**

```
[{
      keyword: 'type',
      dataPath: ".properties['b'].minimum",
      schemaPath: '#/properties/minimum/type',
      params: { type: 'number' },
      message: 'should be number'
}]
```

**What results did you expect?**

If using the meta-schema which allows $data keyword, then schemas which use this keyword should also be valid.
Note that using the schema itself to validate the data will validate correctly.

**Are you going to resolve the issue?**

**Comment by EtaiG** [comment]
I now see that if instead I do this:

```javascript
const Ajv = require('ajv');
const ajv = new Ajv({$data: true});
const isValid = ajv.validateSchema(schema);
```

it works.

However, if I needed to modify the meta schema, it does not work, since I pass `meta: false`

I see I can try to use the $dataMetaSchema method on the constructor as follows:
```javascript
const Ajv = require('ajv');
const validator = new Ajv({$data: true, meta: false});
validator.addMetaSchema(
    Ajv.$dataMetaSchema(draft, ['/properties'])
)
```

but it's annoying that I need to add this parameter `['/properties']`, which I only found through debugging the code.

Is what I'm doing the right way to do this, and if so, is there a reason to need this extra parameter? Would it ever actually be possible to put something other than `/properties` there and have it be meaningful? Or perhaps can this be a default?
